src/main.rs

```diff
@@ -0,0 +1,184 @@
+use std::{env, fs::File, io::Write};
+
+use async_graphql::{
+    extensions::Logger, http::GraphiQLSource, EmptySubscription, SDLExportOptions, Schema,
+};
+
+use async_graphql_axum::{GraphQLRequest, GraphQLResponse};
+
+use axum::{
+    extract::State,
+    http::header::HeaderMap,
+    response::{self, IntoResponse},
+    routing::{get, post},
+    Router, Server,
+};
+
+use clap::{arg, command, Parser};
+
+use simple_logger::SimpleLogger;
+
+use log::info;
+use mongodb::{options::ClientOptions, Client, Database};
+
+mod order;
+use order::Order;
+
+mod order_item;
+
+mod query;
+use query::Query;
+
+mod mutation;
+use mutation::Mutation;
+
+use foreign_types::{Coupon, ProductVariant, ProductVariantVersion, ShipmentMethod, TaxRate};
+
+mod user;
+use user::User;
+
+mod http_event_service;
+use http_event_service::{
+    list_topic_subscriptions, on_id_creation_event, on_product_variant_version_creation_event,
+    on_tax_rate_version_creation_event, HttpEventServiceState,
+};
+
+mod authentication;
+use authentication::AuthorizedUserHeader;
+
+mod base_connection;
+mod discount_connection;
+mod foreign_types;
+mod mutation_input_structs;
+mod order_connection;
+mod order_datatypes;
+mod order_item_connection;
+mod product_variant_version_connection;
+
+/// Builds the GraphiQL frontend.
+async fn graphiql() -> impl IntoResponse {
+    response::Html(GraphiQLSource::build().endpoint("/").finish())
+}
+
+/// Establishes database connection and returns the client.
+async fn db_connection() -> Client {
+    let uri = match env::var_os("MONGODB_URI") {
+        Some(uri) => uri.into_string().unwrap(),
+        None => panic!("$MONGODB_URI is not set."),
+    };
+
+    // Parse a connection string into an options struct.
+    let mut client_options = ClientOptions::parse(uri).await.unwrap();
+
+    // Manually set an option.
+    client_options.app_name = Some("Order".to_string());
+
+    // Get a handle to the deployment.
+    Client::with_options(client_options).unwrap()
+}
+
+/// Returns Router that establishes connection to Dapr.
+///
+/// Creates endpoints to define pub/sub interaction with Dapr.
+async fn build_dapr_router(db_client: Database) -> Router {
+    let product_variant_collection: mongodb::Collection<ProductVariant> =
+        db_client.collection::<ProductVariant>("product_variants");
```

Comment:
all these collection identifier strings are used at multiple places
it is possible to put these somewhere as constants to prevent accidential spelling errors?

Reply:
Good idea, but not yet sure how to implement this. I do not like global state and passing all collections as parameters is scary

Reply:
Could be constants, or i write a function that infers the collection names by the type. The naming schema is standardized

Reply:
Is it ok if we put this in an issue for now?

Reply:
imho this should be not thaaat much work (just some constants), but I guess fine for me